File: go/0001.go
package leetcode

func twoSum(nums []int, target int) []int {
	complementToIndex := make(map[int]int)
	for idx, num := range nums {
		compIdx, ok := complementToIndex[target-num]
		if !ok {
			complementToIndex[num] = idx
			continue
		}
		return []int{compIdx, idx}
	}
	return nil
}

File: python/0001.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement_to_index = dict()
        for idx, num in enumerate(nums):
            if num not in complement_to_index:
                complement_to_index[target - num] = idx
            else:
                return [complement_to_index[num], idx]